### PR TITLE
Collision Resolution Unit Tests and bug fixes

### PR DESCRIPTION
**Pull request by Developer-The-Great**
- Created Collision Resolution unit tests for later
- Created some debugging helpers for physics
- Fixed a bug where adding a rigidbodies would cause entities to visually dissapear

### CODE REVIEW

File: legion/engine/physics/colliders/convexcollider.cpp [hunk 1]
```diff
@@ -1,23 +1,25 @@
 #include <physics/colliders/convexcollider.hpp>
 #include <physics/physics_statics.hpp>
-#include <physics/systems/physicssystem.hpp>
-#include <physics/data/convexconvexpenetrationquery.h>
-#include <physics/data/edgepenetrationquery.h>
+#include <physics/data/identifier.h>
```
Comment (Algo-ryth-mix) [inline legion/engine/physics/colliders/convexcollider.cpp]: why with the .h again

File: legion/engine/physics/data/identifier.h [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+#include <core/core.hpp>
+
+namespace legion::physics
+{
+    struct identifier
+    {
+        std::string id = "";
+    };
+
+}
```
Comment (Algo-ryth-mix) [inline legion/engine/physics/data/identifier.h]: this file should be .hpp instead of .h

File: legion/engine/physics/physics_contact.h [hunk 3]
```diff
@@ -7,33 +7,245 @@ namespace legion::physics
...
+			//where M is the 12x12 matrix
+			//[ ma	0	0	0]
+			//[ 0	Ia	0	0]
+			//[ 0	0	mb	0]
+			//[ 0	0	0  Ib]
+
+			//where ma is a 3x3 identity matrix scaled by the mass of the rigidbodyA,
+			//mb is a 3x3 identity matrix scaled by the mass of the rigidbodyB,
+			//where Ia is the 3x3 matrix inertia tensor of the rigidbodyA,
+			//where Ib is the 3x3 matrix inertia tensor of the rigidbodyB,
```
Comment (Algo-ryth-mix) [inline legion/engine/physics/physics_contact.h]: I trust you on this one chief

**Review by Algo-ryth-mix — CHANGES_REQUESTED** [review]

**Review by Algo-ryth-mix — APPROVED** [review]

File: legion/engine/physics/physicsconstants.h [hunk 4]
```diff
@@ -9,4 +9,10 @@ namespace legion::physics::constants
 	static constexpr int contactSolverIterationCount = 10;
 
 	static constexpr int frictionSolverIterationCount = 4;
+
+	static constexpr float faceToFacePenetrationBias = 0.005f;
+
+	static constexpr float faceToEdgePenetrationBias = 0.01f;
+
+	static constexpr float baumgarteCoefficient = 0.25f;
 }
```
Comment (GlynLeine) [inline legion/engine/physics/physicsconstants.h:18]: this file should be .hpp

File: legion/engine/physics/physics_contact.h [hunk 5]
```diff
@@ -7,33 +7,245 @@ namespace legion::physics
...
-		ecs::component_handle<rigidbody> rbInc;
+		math::vec3 RefWorldContact;
+		math::vec3 IncWorldContact;
 
-		math::vec3 worldContactRef;
-		math::vec3 worldContactInc;
+		math::vec3 refRBCentroid;
+		math::vec3 incRBCentroid;
 
 		math::mat4 refTransform;
```
Comment (GlynLeine) [inline legion/engine/physics/physics_contact.h]: this file should be .hpp instead of .h

**Review by GlynLeine — CHANGES_REQUESTED** [review]

**Review by GlynLeine — APPROVED** [review]

**Pull request merged**